lightning/src/onion_message/packet.rs

```diff
@@ -114,11 +114,10 @@ pub(super) enum Payload<T: OnionMessageContents> {
 	}
 }
 
-/// The contents of an [`OnionMessage`] as read from the wire.
-///
-/// [`OnionMessage`]: crate::ln::msgs::OnionMessage
-#[derive(Debug)]
-pub enum ParsedOnionMessageContents<T: OnionMessageContents> {
+// #[derive(Debug)]
+/// The contents of an onion message. In the context of offers, this would be the invoice, invoice
+/// request, or invoice error.
+pub enum OnionMessageContents<T: CustomOnionMessageContents> {
```

Comment:
This looks unintentional from trying to resolve a rebase conflict. Likewise elsewhere. Note that the `CustomOnionMessageContents ` trait is now the more general `OnionMessageContents` and the old `OnionMessageContents` is now called `ParsedOnionMessageContents`. See 94573dda33c2e6bf55b2f28ffe7fbdf0a37f6edc and 840efd533468cf7cacd8b4bab27ce2db86b3eb93.

Reply:
Ok , but here in this function. 
`pub fn send_onion_message<T: OnionMessageContents>(
		&self, path: OnionMessagePath, contents: T, reply_path: Option<BlindedPath>
	) -> Result<(), SendError> {`
Now `contents` is defined here as trait and logging a trait is not possible , how to log `contents` here.
Thanks :)

Reply:
You'll need to add a `Debug

Reply:
You'll need to add a `Debug` trait bound to `T` as mentioned in https://github.com/lightningdevkit/rust-lightning/pull/2642#discussion_r1364108504, i.e. `T: OnionMessageContents + Debug`, or alternatively make `OnionMessageContents` a subtrait of `Debug`.

Reply:
We'll probably want the latter so we don't need to add a bound in multiple places (i.e. both on `ParsedOnionMessageContents` and `send_onion_message` vs just on `OnionMessageContents`. So, it would be:

```rust
pub trait OnionMessageContents: Writeable + Debug {
```